### ISSUE TITLE
TypeError: The "listener" argument must be of type Function. Recei	ved type function

### ISSUE THREAD

**Issue by JohnCido** [vercel/storage#323]
I am using `@vercel/postgres` with `drizzle-orm` in my project. Everything works fine when I run my project on localhost, but when I try to use the deployed online version, the following error was throwed whenever I try to perform any db operations:
```console
TypeError: The "listener" argument must be of type Function. Recei	ved type function
    at checkListener (/var/task/vercel/path0/apps/web/dist/index.cjs:162444:12)
    at _addListener (/var/task/vercel/path0/apps/web/dist/index.cjs:162462:15)
    at hn.addListener (/var/task/vercel/path0/apps/web/dist/index.cjs:162469:79)
    at new Connection (/var/task/vercel/path0/apps/web/dist/index.cjs:163248:21)
    at new Client (/var/task/vercel/path0/apps/web/dist/index.cjs:163284:16)
    at new NeonClient (/var/task/vercel/path0/apps/web/dist/index.cjs:163591:156)
    at new VercelClient (/var/task/vercel/path0/apps/web/dist/index.cjs:163752:20)
    at VercelPool.newClient (/var/task/vercel/path0/apps/web/dist/index.cjs:163429:75)
    at VercelPool.connect (/var/task/vercel/path0/apps/web/dist/index.cjs:163429:24)
    at VercelPool.connect (/var/task/vercel/path0/apps/web/dist/index.cjs:163803:18)
```

The framwork I'm using is SolidStart. The website is inside a turborepo.

This is my Drizzle setup if it helps:
```ts
import {
  drizzle,
} from 'drizzle-orm/vercel-postgres'
import {
  createPool,
} from '@vercel/postgres'

const client = createPool({
  connectionString: process.env.SERVER_DB_URL,
})

export const db = drizzle(client, {
  schema,
  logger: IS_DEV,
})
```

**Comment by correttojs** [comment]
hi @JohnCido, I suggest to use the `createClient` method instead of `createPool`.

I'm not able to reproduce the issue using `createPoll` anyway, could you share a full repo reproducing the issue?

**Comment by fredgig** [comment]
I ran into the same issue using a similar setup. With the way `neondatabase/serverless` is bundled, the build output of your solid-start application located at `.vercel/output/functions/.../index.(mjs/cjs)` (assuming you're using the Vercel adapter) will add a tab to the beginning of each line, making the code below evaluate `typeof r!="functi\ton"` rather than `typeof r!="function"`. This is the same reason there's a strange whitespace in the word `Recei ved` in the error message.

```js
	0;B.prototype._maxListeners=void 0;var zn=10;function ut(r){if(typeof r!="functi\
	on")throw new TypeError('The "listener" argument must be of type Function. Recei\
	ved type '+typeof r)}a(ut,"checkListener");Object.defineProperty(B,"defaultMaxLi\
```

I'm not quite sure how to resolve this issue because I'm a complete stranger to vite/esbuild notions, but perhaps that could put you on the right track.

tldr I don't think this is an issue specific to this repository.

**Comment by fredgig** [comment]
Turns out this is an issue caused by `@rollup/plugin-commonjs`. You can avoid it by setting the `build.commonjsOptions.strictRequires` value to `false` in your vite.config.ts:

```ts
// ...

export default defineConfig({
	// ...
	build: {
		commonjsOptions: {
			strictRequires: false
		}
	}
});

```

**Comment by vvo** [comment]
Thanks! We had a few issues with the Solid adapter, perhaps they will need to be opened on the solid repository to have them fixed completely.
